app/Http/Controllers/BookingAPIController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Booking;
+use App\Models\Room;
+use Carbon\CarbonPeriod;
+use http\Env\Response;
+use Illuminate\Http\Request;
+
+class BookingAPIController extends Controller
+{
+    public function create(Request $request)
+    {
+
+        $request->validate([
+            'room_id' => 'required|integer|exists:rooms,id',
+            'customer' => 'required|string|max:255',
+            'guests' => 'required|integer|min:1',
+            'start' => 'required|date|date_format:Y-m-d|after_or_equal:today',
+            'end' => 'required|date|date_format:Y-m-d|after:start',
+        ]);
+
+        $room = Room::find($request->room_id);
+        $startDate = $request->start;
+        $endDate = $request->end;
```

Comment:
I would actually suggest not using these variables, and instead using $request directly when you need the dates. 

While having the variables is less typing, it does make it less obvious what the $startDate and $endDate actually are - whereas having $request->start makes it super clear that it's the start that the user sent in their request